src/ValidationsFactory.js

```diff
@@ -0,0 +1,195 @@
+      this.screen.pagesValidated.push(parseInt(this.element.config.eventData));
+      if (this.screen.config[this.element.config.eventData] && this.screen.config[this.element.config.eventData].items) {
+        await ValidationsFactory(this.screen.config[this.element.config.eventData].items, { screen: this.screen }).addValidations(validations);
+      }
+    }
+  }
+}
+
+/**
+ * Add validations for a form element
+ */
+class FormElementValidations extends Validations {
+  async addValidations(validations) {
+    if (!(this.element.config && this.element.config.name && typeof this.element.config.name === 'string' && this.element.config.name.match(/^[a-zA-Z_][0-9a-zA-Z_.]*$/))) {
+      //element invalid
+      return;
+    }
+    const fieldName = this.element.config.name;
+    const validationConfig = this.element.config.validation;
+
+    set(validations, fieldName, get(validations, fieldName, {}));
+    const fieldValidation = get(validations, fieldName);
+    if (validationConfig instanceof Array) {
+      validationConfig.forEach((validation) => {
+        const rule = this.camelCase(validation.value.split(':')[0]);
+        if (!rule) {
+          return;
+        }
+        let validationFn = validators[rule];
+        if (!validationFn) {
+          // eslint-disable-next-line no-console
+          console.error(`Undefined validation rule "${rule}"`);
+          return;
+        }
+        if (validation.configs instanceof Array) {
+          const params = [];
+          validation.configs.forEach((cnf) => {
+            params.push(cnf.value);
+          });
+          validationFn = validationFn(...params);
+        }
+        fieldValidation[rule] = validationFn;
+      });
+    } else if (typeof validationConfig === 'string' && validationConfig) {
+      let validationFn = validators[validationConfig];
+      if (!validationFn) {
+        // eslint-disable-next-line no-console
+        console.error(`Undefined validation rule "${validationConfig}"`);
+        return;
+      }
+      fieldValidation[validationConfig] = validationFn;
+    }
+    if (this.element.items) {
+      ValidationsFactory(this.element.items, { screen: this.screen }).addValidations(validations);
+    }
+  }
+  camelCase(name) {
+    return name.replace(/_\w/g, m => m.substr(1, 1).toUpperCase());
+  }
+}
+
+function ValidationsFactory(element, options) {
+  if (element instanceof Array) {
+    return new ArrayOfFieldsValidations(element, options);
+  }
+  if (element.config instanceof Array) {
+    return new ScreenValidations(element, options);
+  }
+  if (element.component === 'FormNestedScreen') {
+    return new FormNestedScreenValidations(element, options);
+  }
+  if (element.component === 'FormMultiColumn') {
+    return new FormMultiColumnValidations(element, options);
+  }
+  if (element.component === 'FormLoop') {
+    return new FormLoopValidations(element, options);
+  }
+  if (element.component === 'FormRecordList') {
```

Comment:
Remove the commented code or the complete block if it is not used

Reply:
solved